applications/sol2ink-follow-up.md

```diff
@@ -0,0 +1,248 @@
+
+* **Total Estimated Duration:** 10 weeks
+* **Full-Time Equivalent (FTE):** 2-2.5
+* **Total Costs:** 60,000 USD
+
+
+### Previous work
+
+### Milestone 1 — Implement	Sol2Ink cli tool for simple contract parsing
+
+- **Estimated duration:** 8 weeks
+- **FTE:**  1.5
+- **Costs:** 36,000 USD
+- **Delivery status:** [Delivered](https://github.com/w3f/Grant-Milestone-Delivery/blob/master/deliveries/sol2ink-delivery.md)
+
+| Number | Deliverable | Specification |
+| -----: | ----------- | ------------- |
+| 0a. | License | MIT |
+| 0b. | Documentation | We will provide a documentation for the user on how to use our tool for simplification of their process of transformation of Solidity code to Rust and Ink! code, along with inline documentation of code so the developers can better understand the processes in the backend |
+| 1. | Sol2Ink CLI | A CLI application which will take a Solidity file as the input and produce a transformed Rust file with Rust and ink! code as the output |
+| 2. | Integrate OpenBrush | Since OpenBrush is a tool to ease and fasten the smart contracts development in Ink!, we will add this library to the smart contracts generated by Sol2Ink |
+| 3. | Website with guides | We will provide a website where we will compare a transformation of Solidity code into Rust and ink! code, along with issues which may occur (for example handling of Assembly blocks etc.). |
+
+
+### Current work - Scope of this Grant
+
+
+### Milestone 2 — Upgrade focused on complex Smart contract applications parsing
+
+- **Estimated duration:** 10 weeks
+- **FTE:**  2-2.5
+- **Costs:** 60,000 USD
+
+| Number | Deliverable | Specification |
+| -----: | ----------- | ------------- |
+| 0a. | License | MIT |
+| 0b. | Documentation | We will enhance both the inline documentation as well as the website with guides to reflect new features implemented in this milestone |
+| 0c. | Testing and Testing guide | Core functions will be fully covered by comprehensive unit tests to ensure functionality and robustness. Documentation will describe how to run these tests. |
+| 0d. | Docker | We will provide a Dockerfile(s) that can be used to test all the functionality delivered with this milestone. |
+| 1. | Website with guides | We will enhance the existing guides to reflect new features implemented in this milestone |
+| 2. | Split contract into traits | Currently all logic and storage is in one contract file. This upgrade will generate traits for each contract that the contract is using, splitting logic and storage into separate traits. This will also nicely handle inheritance of Solidity smart contracts |
+| 3. | Sol2Ink transpiler update | We will update the transpiler to deal with code which it currently does not support or parses incorrectly |
+| 4. | Library parsing | We will implement the parser for Solidity libraries |
```

Comment:
Does this concern the input? If yes, what's the difference between importing from a solidity contract directly vs. importing from a library? Isn't the input in both cases just a solidity file?

Reply:
its just that currently sol2ink looks for contract/ interface definition, it skips library for now because we want to deal with it in a different way than with contracts, meaning if you change the library to a contract, it will be parsed, butnsince a library is not a contract we will have to do it in a different way

Reply:
Thanks, could you integrate this information into the deliverable spec?